back/engines/commercial/analytics/app/services/analytics/query_runner_service.rb

```diff
@@ -31,8 +31,11 @@ def run(query)
         results = query_order(results)
       end
 
-      limit = @json_query.fetch(:limit, 1000)
-      results = results.limit(limit)
+      if @json_query.key?(:page)
```

Comment:
why is there backend stuff here?

Reply:
Because I need the pagination logic, and that hasn't been released yet

Reply:
yeah, sorry i should have elaborated..
why aren't we doing as previous ocasions where the front end branch was branching from a backend branch?

Reply:
Not sure, I guess I started with this branch first and only halfway through we realized that we needed this bit of backend code. So James wrote it and I merged it into this branch. Is there anything against doing it like this?

Reply:
ok maybe is fine, im going to be working on the backend for the pagination so we might want to merge that here. but lets discuss it on slack